src/core/mac/mac_frame.hpp

```diff
@@ -512,15 +513,40 @@ class Frame : public otRadioFrame
      * @returns The Sequence Number value.
      *
      */
-    uint8_t GetSequence(void) const { return GetPsdu()[kSequenceIndex]; }
+    uint8_t GetSequence(void) const
+    {
+        OT_ASSERT(IsSequencePresent());
```

Comment:
There is a code size increase of ~300 bytes. Is this due to the asserts? If so, do we really need them?

Reply:
I tried removing it and it seems it doesn't help a lot.

Reply:
I checked the details size report in https://github.com/openthread/openthread/actions/runs/10174568011/job/28140508466?pr=10544, I didn't see an outstanding abnormal size increase.